### PR TITLE
584 dexbot installer

### PR DESCRIPTION
**Pull request by dominic22**
Added dexbot installer and integrated it into the appveyor build process. Furthermore the new icon of dexbot and an application icon were added.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -11,28 +11,37 @@ environment:
...
   - make install
-
-after_test:
   - make package
+  - copy /Y C:\projects\dexbot\dist\DEXBot-gui.exe C:\projects\dexbot\installer\windows\msi\DEXBot.exe
   - '7z a DEXBot-gui-win64.zip %APPVEYOR_BUILD_FOLDER%\dist\DEXBot-gui.exe'
+  # read dexbot version from __init__.py
+  - for /f "delims== eol=# tokens=1,2" %%G in (.\dexbot\__init__.py) do set %%G=%%H
+  # set to version and replace ' with empty character
+  - set ApplicationVersion=%VERSION:'=%
```
Comment (joelvai) [inline appveyor.yml]: If you can get the version number, add it to the distribution name above as well like `DEXBot-gui-win64-VERSION_NUMBER.zip`

File: appveyor.yml [hunk 2]
```diff
@@ -11,28 +11,37 @@ environment:
...
+- Release
+platform: x86
+build:
+  verbosity: normal
 
 # @TODO: Run tests..
 test_script:
   - "echo tests..."
 
+# artifact as relativ path to the build folder C:\projects\dexbot
```
Comment (joelvai) [inline appveyor.yml]: Fix typo, and start all comments as capitalized

File: appveyor.yml [hunk 3]
```diff
@@ -11,28 +11,37 @@ environment:
...
 install:
   - SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%;C:\MinGW\bin
   - copy c:\MinGW\bin\mingw32-make.exe c:\MinGW\bin\make.exe
   - copy c:\Python36-x64\python.exe c:\Python36-x64\python3.exe
   - python --version
   - make install
-
-after_test:
   - make package
+  - copy /Y C:\projects\dexbot\dist\DEXBot-gui.exe C:\projects\dexbot\installer\windows\msi\DEXBot.exe
```
Comment (joelvai) [inline appveyor.yml]: Fix hard coded paths using `%APPVEYOR_BUILD_FOLDER%` (like below), and is this necessary at all?

File: installer/windows/bundle/resources/variables.wxi [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Include>
+  <?if $(var.ApplicationVersion) = ""?>
+  <?define ApplicationVersion = "0.10.0" ?>
```
[the remark this answers is not in the document]
Comment (dominic22) [reply installer/windows/bundle/resources/variables.wxi]: It is a placeholder for local development but overwritten in the appveyor build script by the current version specified in __init__.py

File: appveyor.yml [hunk 5]
```diff
@@ -11,28 +11,37 @@ environment:
...
-configuration: Release
-
 install:
   - SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%;C:\MinGW\bin
   - copy c:\MinGW\bin\mingw32-make.exe c:\MinGW\bin\make.exe
   - copy c:\Python36-x64\python.exe c:\Python36-x64\python3.exe
   - python --version
   - make install
-
-after_test:
```
[the remark this answers is not in the document]
Comment (dominic22) [reply appveyor.yml]: I moved it to install since i need the executable before the visual studio build process

File: appveyor.yml (hunk 1 above)
Comment (dominic22) [reply appveyor.yml]: Ok I will add this, shall I add the version to the installer as well?

Comment (joelvai) [reply appveyor.yml]: Do that. :+1:

**Comment by bitfag** [comment]
Binary files to remove from history to prevent unneeded repo growth:

installer/windows/bundle/prerequisites/python-3.7.3.exe
installer/windows/msi/DEXBot.exe

**Comment by bitfag** [comment]
Hmm, looks like installer/windows/bundle/prerequisites/python-3.7.3.exe is listed installer/windows/bundle/bundle.wixproj as a requirement (it's needed to build the installer). It's a generally bad idea to put such binaries in repo, we should consider moving this binary outside and probably download it at build time.

**Comment by bitfag** [comment]
#626

**Pull request closed without merging**